core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -243,9 +243,15 @@ open class TileInfo {
         if (hasViewableResource(observingCiv) && getTileResource().improvement == improvement.name)
             stats.add(getTileResource().improvementStats!!.clone()) // resource-specific improvement
 
+        // As of 3.10.5 This is to be deprecated and converted to "[stats] once [tech] is discovered" - keeping it here to that mods with this can still work for now
         if (improvement.improvingTech != null && observingCiv.tech.isResearched(improvement.improvingTech!!))
             stats.add(improvement.improvingTechStats!!) // eg Chemistry for mines
 
+        for (unique in improvement.uniqueObjects) when (unique.placeholderText) {
+            "[] once [] is discovered" -> if (observingCiv.tech.isResearched(unique.params[1])) stats.add(Stats.parse(unique.params[0]))
+            "[] on fresh water tiles once [] is discovered" -> if (observingCiv.tech.isResearched(unique.params[1]) && isAdjacentToFreshwater) stats.add(Stats.parse(unique.params[0]))
```

Comment:
"[] on [] tiles once [] is discovered"
then "fresh water" and "non-fresh water tiles" are parameters, and "[] tiles" can be united with the other tile parameter check function

Reply:
unified